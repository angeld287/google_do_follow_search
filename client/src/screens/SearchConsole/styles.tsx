const styles = {
    container: {
        width: '70%',
        margin: "50px auto",
        backgroundColor: "white",
        flexDirection: "column",
        justifyContent: "center",
        alignItems: "center",
    } as React.CSSProperties,
    buttons: {
        textAlign: 'right'
    } as React.CSSProperties,
    button: {
        marginRight: 5
    } as React.CSSProperties,
};

export default styles;