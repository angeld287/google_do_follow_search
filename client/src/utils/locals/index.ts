// @ts-nocheck
/**
 * Define App Locals & Configs
 *
 * @author Angel Angeles <aangeles@litystyles.com>
 */

class Locals {

    /**
     * Makes env configs available for your app
     * throughout the app's runtime
     */
    public static config(): any {
        const server_host = "52.70.36.16";
        //(
        //    window._env_ ?
        //        window._env_.REACT_APP_SERVER_HOST : (process.env.SERVER_HOST || "localhost"));

        const server_port = "3000"
        //(
        //    window._env_ ?
        //        window._env_.REACT_APP_SERVER_PORT : (process.env.REACT_APP_SERVER_PORT || 3000));

        const server_url = "http://" + server_host + ":" + server_port

        const a_tags_reg_exp = /(<a\s*(?!.*\bnofollow)[^>]*)(href="https?:\/\/)((?!(?:(?:www\.)?'.implode('|(?:www\.)?', $follow_list).'))[^"]+)"((?!.*\bnofollow)[^>]*)(?:[^>]*)>/g
        const follow_reg_exp = /rel=("|')dofollow("|')/g
        const no_follow_reg_exp = /rel=("|')nofollow("|')/g

        return {
            server_url,
            a_tags_reg_exp,
            follow_reg_exp,
            no_follow_reg_exp
        }
    }
}

export default Locals;