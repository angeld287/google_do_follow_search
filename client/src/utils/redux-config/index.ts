import { configureStore } from "@reduxjs/toolkit";
import userSessionReducer from '../../features/userSession/userSessionSlice';
import userRegisterReducer from '../../features/userRegister/userRegisterSlice';
import userSearchSlice from '../../features/googleSearch/searchSlice';

export const store = configureStore({
    reducer: {
        userSession: userSessionReducer,
        userRegister: userRegisterReducer,
        googleSearch: userSearchSlice,

    },
    middleware: (getDefaultMiddleware) => getDefaultMiddleware({
        serializableCheck: false
    })
});