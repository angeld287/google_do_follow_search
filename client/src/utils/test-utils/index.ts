// test-utils.js
import { render, queries, RenderOptions, RenderResult } from '@testing-library/react'
import { JSXElementConstructor, ReactElement } from 'react';
import * as customQueries from './custom-queries'
import * as customFunctionalities from './functionalities';

const customRender = (ui: ReactElement<any, string | JSXElementConstructor<any>>, options?: RenderOptions) =>
    render(ui, { queries: { ...queries, ...customQueries, ...customFunctionalities }, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }