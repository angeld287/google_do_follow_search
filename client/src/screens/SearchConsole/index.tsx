import { DoubleLeftOutlined, DoubleRightOutlined } from '@ant-design/icons';
import { Alert } from 'antd';
import { Content } from 'antd/lib/layout/layout';
import React, { useCallback, useState } from 'react';
import { useAppDispatch, useAppSelector } from '../../app/hooks';
import CustomButton from '../../components/CustomButton';
import CustomList from '../../components/CustomList';
import CustomSearch from '../../components/CustomSearch';
import { searchAsync } from '../../features/googleSearch/asyncThunks';
import { selectSearch } from '../../features/googleSearch/searchSlice';
import styles from './styles';

const SearchConsole: React.FC = () => {

    const session = useAppSelector(selectSearch);
    const dispatch = useAppDispatch()
    const [error, setError] = useState(false);
    const [index, setIndex] = useState(0);
    const [value, setValue] = useState("");

    const onSearch = useCallback((text: string) => {
        setError(false)
        setIndex(0)
        if (text === '') {
            setError(true)
            return null
        }

        setValue(text)
        dispatch(searchAsync({ text, index: 0 }))
    }, [dispatch]);

    const getPreviousItems = useCallback(() => {
        let _index = index - 10;
        setIndex(_index)
        dispatch(searchAsync({ text: value, index: _index }))
    }, [dispatch, value, index])

    const getNextItems = useCallback(() => {
        let _index = index + 10;
        setIndex(_index)
        dispatch(searchAsync({ text: value, index: _index }))
    }, [dispatch, value, index])

    return (
        <Content style={styles.container}>
            <h1>G🅾️🅾️🇬le 🔙Links 🔗 👉 SEARCH 🔍 📙</h1>
            <CustomSearch placeholder='Type the long keyword' onSearch={onSearch} loading={session.status === 'pending'} />
            {
                error &&
                (
                    <div style={{ textAlign: 'right' }}>
                        <Alert message="Please type some keyword." type="error" />
                    </div>
                )
            }
            {(session.results.length !== 0) &&
                <>
                    <CustomList data={session.results} />
                    <div style={styles.buttons}>
                        <CustomButton onClick={getPreviousItems} icon={<DoubleLeftOutlined />} customStyle={styles.button} htmlType="button" _key="btn-previous" disabled={(session.results.length === 0 || index === 0)} color="blue">Previous</CustomButton>
                        <CustomButton onClick={getNextItems} customStyle={styles.button} htmlType="button" _key="btn-next" disabled={session.results.length === 0} color="blue">Next <DoubleRightOutlined /></CustomButton>
                    </div>
                </>
            }

        </Content>
    );
};

export default React.memo(SearchConsole);