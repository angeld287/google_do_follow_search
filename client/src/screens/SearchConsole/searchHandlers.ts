import { headers, json, _rest } from "../../utils/test-handlers-utils";

const search = _rest.post('http://localhost:3000/api/search', (req, res, ctx) => {
    const _headers = headers(ctx);
    let _json = json(ctx, {});

    let _result: any = null;

    _result = {
        success: true,
        results: [
            { title: 'Test 2', position: 2 },
            { title: 'Test 1', position: 1 },
            { title: 'Test 3', position: 3 },
            { title: 'Test 4', position: 4 },
            { title: 'Test 5', position: 5 },
            { title: 'Test 6', position: 6 },
            { title: 'Test 7', position: 7 },
            { title: 'Test 8', position: 8 },
            { title: 'Test 9', position: 9 },
            { title: 'Test 10', position: 10 },
        ],
    }

    _json = json(ctx, {
        statusCode: "10000", message: "Success",
        data: _result
    });

    return res(..._headers, _json, ctx.delay(10))
});

const handlers = [search]

export default handlers