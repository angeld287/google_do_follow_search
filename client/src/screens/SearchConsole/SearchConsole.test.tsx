import React from 'react';
import { render, RenderResult, functions, fireEvent, screen, waitFor, act } from '../../utils/test-utils';
import { Provider } from 'react-redux';
import { setupServer } from 'msw/node'
import handlers from './searchHandlers';

import SearchConsole from './index';
import { store } from '../../utils/redux-config';

let component: RenderResult;

const server = setupServer(...handlers)

describe("Search Console Test Suite", () => {
    beforeAll(() => server.listen({
        onUnhandledRequest: ({ headers, method, url }) => {
            if (headers.get("User-Agent") !== "supertest") {
                throw new Error(`Unhandled ${method} request to ${url}`);
            }
        },
    }));

    beforeEach(() => {
        if (component)
            component.unmount()

        act(() => {
            component = render(
                <Provider store={store}>
                    <SearchConsole />
                </Provider>
            )
        });
        server.resetHandlers()
    });

    afterAll(() => server.close())

    test('It must render the component correcly', () => {
        expect(component.asFragment()).toMatchSnapshot();
    })

    test('It must display basic Search Console components', () => {
        expect(component.getByPlaceholderText(/Type the long keyword/i)).toBeInTheDocument();
        expect(component.getAllByText(/Search/i)[0]).toBeInTheDocument();
    });

    test('It must respond "Please type some keyword." when the input is blank.', async () => {

        functions.writeInInputFoundByPlaceHolder(null, /Type the long keyword/i, "");

        await act(() => {
            fireEvent.click(screen.getAllByText(/Search/i)[1]);
        });

        await waitFor(() => {
            component.rerender(
                <Provider store={store}>
                    <SearchConsole />
                </Provider>
            )
            expect(screen.getByText("Please type some keyword.")).toBeInTheDocument();
        });
    });
});