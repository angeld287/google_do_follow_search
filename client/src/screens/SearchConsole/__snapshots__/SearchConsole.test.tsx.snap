// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Console Test Suite It must render the component correcly 1`] = `
<DocumentFragment>
  <main
    class="ant-layout-content"
    style="width: 70%; margin: 50px auto; background-color: white; flex-direction: column; justify-content: center; align-items: center;"
  >
    <h1>
      G🅾️🅾️🇬le Do Follow 👉 SEARCH 🔍 📙
    </h1>
    <span
      class="ant-input-group-wrapper ant-input-search ant-input-search-large ant-input-search-with-button ant-input-group-wrapper-lg"
    >
      <span
        class="ant-input-wrapper ant-input-group"
      >
        <span
          class="ant-input-affix-wrapper ant-input-affix-wrapper-lg"
        >
          <input
            class="ant-input ant-input-lg"
            placeholder="Type the long keyword"
            type="text"
            value=""
          />
          <span
            class="ant-input-suffix"
          >
            <span
              class="ant-input-clear-icon ant-input-clear-icon-hidden"
              role="button"
              tabindex="-1"
            >
              <span
                aria-label="close-circle"
                class="anticon anticon-close-circle"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="close-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                  />
                </svg>
              </span>
            </span>
          </span>
        </span>
        <span
          class="ant-input-group-addon"
        >
          <button
            class="ant-btn ant-btn-primary ant-btn-lg ant-input-search-button"
            type="button"
          >
            <span>
              Search
            </span>
          </button>
        </span>
      </span>
    </span>
  </main>
</DocumentFragment>
`;
